src/hypothesis/extra/datetime.py

```diff
@@ -123,3 +123,34 @@ def times(allow_naive=None, timezones=None):
 
 def datetime_to_time(dt):
     return dt.timetz()
+
+
+class TimedeltaStrategy(SearchStrategy):
+
+    def __init__(self, min_value=dt.timedelta.min, max_value=dt.timedelta.max):
+        assert type(min_value) == dt.timedelta, 'min_value must be a timedelta'
+        assert type(max_value) == dt.timedelta, 'min_value must be a timedelta'
+        assert min_value <= max_value,\
```

Comment:
Better: use the `check_valid_interval` function from `hypothesis.strategies`, which exposes a nicer error message. Look at the [implementation of `integers()`](https://github.com/HypothesisWorks/hypothesis-python/blob/master/src/hypothesis/strategies.py#L193) for an example.